Open pull request with changes from an active fork

brtkwr opened the pull request.

dougszumski committed:
Extend arguments passed to Monasca setup

This patch extends the number of arguments which can be passed
to Monasca setup. The additional arguments may be required by
some operators to correctly configure the Monasca agent for their
environment.

dougszumski committed:
Use double braces for expressions

To prevent these variables being interpreted as strings wrap them
in double braces.

dougszumski committed:
Upgrade setuptools in venv

Recent versions of the Monasca agent require a minimum of
setuptools 1.4. If an earlier version is installed the installation
of the Monasca agent fails, complaining that it can't upgrade
setuptools, and that version >=1.4 is required.

An alternative would be to support installing arbitrary pip
packages into the venv via a user configurable variable.

dougszumski committed:
Merge pull request #1 from stackhpc/bugfix/upgrade_setuptools

Upgrade setuptools in venv

dougszumski committed:
Merge pull request #3 from stackhpc/feature/extend_configuration_options

Extend arguments passed to Monasca setup

dougszumski committed:
Merge pull request #2 from stackhpc/bugfix/wrap_vars_in_braces

Use double braces for expressions

dougszumski committed:
Update Galaxy meta

dougszumski committed:
Update README

dougszumski committed:
Add support for SuSE

dougszumski committed:
Support specifying a user with which to run the agent

This is used in the systemd unit file. If it's not specified
monasca setup uses mon-agent by default.

dougszumski committed:
Add python-virtualenv as a dependency

dougszumski committed:
Merge pull request #4 from stackhpc/bugfix/add_missing_dependency

Add python-virtualenv as a dependency

dougszumski committed:
Support installing plugins from a role variable

The role variable monasca_agent_custom_plugin_repos, should contain
a list of git repos holding plugins.

Each repo should be configured to install the plugins to the correct
location on the host. See https://github.com/openstack/monasca-agent/
blob/master/docs/Customizations.md#adding-custom-check-plugins

dougszumski committed:
Merge pull request #5 from stackhpc/feature/custom_plugins_from_repo

Support installing plugins from a role variable

dougszumski committed:
Install python-pip for installing agent plugins

dougszumski committed:
Merge pull request #6 from stackhpc/bugfix/install_python_pip

Install python-pip for installing agent plugins

dougszumski committed:
Support installing from git repo

Monasca-agent versions are not tied to specific git branches. This
change supports installing from specific git branches.

dougszumski committed:
Merge pull request #7 from stackhpc/feature/support_install_from_git

Support installing from git repo

oneswig committed:
Install latest versions of pip and setuptools into Monasca-agent virtualenv

dougszumski committed:
Merge pull request #8 from stackhpc/latest-pip

Install latest versions of pip and setuptools

Commit:
Add travis config

Commit:
Support custom git repo

Commit:
Merge pull request #11 from stackhpc/custom-git-repo

Support custom git repo

Commit:
Merge pull request #10 from stackhpc/travis

Add travis config

Commit:
Support pip extra args

Commit:
Update main.yml

Commit:
Update main.yml

Commit:
Fix platforms style in meta/main.yml for successful import to galaxy

dougszumski committed:
Add some repos missing on minimal spins

These packages may be missing on minimal distros. Tested
on Ubuntu and Centos. Suse packages were checked to exist.

dougszumski committed:
Merge pull request #13 from stackhpc/bugfix/more_packages

Add some repos missing on minimal spins

dougszumski committed:
Ensure that plugin dependencies are availabe in Monasca Agent venv

dougszumski committed:
Merge pull request #14 from stackhpc/bugfix/dependencies

Ensure that plugin dependencies are availabe in Monasca Agent venv

dougszumski committed:
Support skipping detection of plugins

Users may not always want detection of a certain plugin to
happen. This can be for example when custom plugin config
is applied, which you do not want to be overwritten, or
when you do not want a particular plugin to be enabled.

dougszumski committed:
Merge pull request #15 from stackhpc/feature/support_skip_detection

Support skipping detection of plugins

dougszumski committed:
Workaround for custom Monasca Agent plugins

We need to install the Monasca Agent custom plugins in the same venv
as the Monasca Agent so that the required dependencies are available
when the Agent runs the plugins. However, the Monasca Agent searchs
a hard coded path for plugins outside of the venv. This change symlinks
the plugin folder from the venv to the hard coded path.

Eventually the Monasca Agent should be able to discover the plugins
from the venv.

dougszumski committed:
Merge pull request #16 from stackhpc/plugin_workaround

Workaround for custom Monasca Agent plugins

dougszumski committed:
Support setting Monasca statsd aggregation interval

Depends on: https://review.openstack.org/#/c/638423/

dougszumski committed:
Merge pull request #17 from stackhpc/feature/statsd_agg

Support setting Monasca statsd aggregation interval

dougszumski committed:
Fix typo about host alive check config

dougszumski committed:
Merge pull request #18 from stackhpc/bugfix/typo

Fix typo about host alive check config

Commit:
Use monasca-agent.target otherwise systemd gets confused

Commit:
Make it backward compatiable for versions prior to Rocky

Commit:
Rename unit file names and also delete old directory to replace with symlink

Commit:
Delete old unit file

brtkwr committed:
Merge pull request #19 from stackhpc/target

Use monasca-agent.target otherwise systemd gets confused

Commit:
Verbose output about what unit file is going to be used

pescobar committed:
install lshw as dependency

dougszumski committed:
Merge pull request #20 from pescobar/install_lshw

install lshw as dependency

dougszumski committed:
Let agent requirements manage psutil version

Many aeons ago, psutil wasn't in global requirements, but now it is.

dougszumski committed:
Merge pull request #21 from stackhpc/remove_decrepit_psutil_from_venv

Let agent requirements manage psutil version

dougszumski committed:
Tweak support for extra args

Builds on existing change. Useful for upper constraints.

dougszumski committed:
Fix issues with no custom plugin repos

If list isn't defined, role fails.

brtkwr committed:
Merge pull request #12 from stackhpc/pip-extra-args

Support pip extra args

witekest commented:
Nice to see you keeping this repository up-to-date. Have you considered contributing it to OpenStack Ansible?

oneswig committed:
Lint pedantry: spaces around doube-brace expansions

oneswig committed:
CentOS is not an OS choice as far as Ansible meta is concerned.

oneswig committed:
Merge pull request #22 from stackhpc/lint-fixes

Lint fixes

oneswig committed:
Add configuration necessary to install monasca-agent on CentOS 8

Different method for managing package name dependencies to enable
a distinction between CentOS 8 and CentOS 7.

oneswig committed:
Feedback nit from Doug

oneswig committed:
Check for python version in venv site-packages path

dougszumski committed:
Merge pull request #23 from stackhpc/centos8-oneswig

Add configuration necessary to install monasca-agent on CentOS 8
